lightning/src/ln/features.rs

```diff
@@ -262,36 +397,33 @@ impl<T: sealed::Context> Features<T> {
 
 impl<T: sealed::DataLossProtect> Features<T> {
 	pub(crate) fn supports_data_loss_protect(&self) -> bool {
-		self.flags.len() > 0 && (self.flags[0] & 3) != 0
+		<T as sealed::DataLossProtect>::supports_feature(&self.flags)
 	}
 }
 
 impl<T: sealed::UpfrontShutdownScript> Features<T> {
 	pub(crate) fn supports_upfront_shutdown_script(&self) -> bool {
-		self.flags.len() > 0 && (self.flags[0] & (3 << 4)) != 0
+		<T as sealed::UpfrontShutdownScript>::supports_feature(&self.flags)
 	}
 	#[cfg(test)]
-	pub(crate) fn unset_upfront_shutdown_script(&mut self) {
-		self.flags[0] ^= 1 << 5;
+	pub(crate) fn clear_upfront_shutdown_script(mut self) -> Self {
+		<T as sealed::UpfrontShutdownScript>::clear_bits(&mut self.flags);
+		self
 	}
 }
 
 impl<T: sealed::VariableLengthOnion> Features<T> {
 	pub(crate) fn supports_variable_length_onion(&self) -> bool {
-		self.flags.len() > 1 && (self.flags[1] & 3) != 0
+		<T as sealed::VariableLengthOnion>::supports_feature(&self.flags)
 	}
 }
 
 impl<T: sealed::InitialRoutingSync> Features<T> {
 	pub(crate) fn initial_routing_sync(&self) -> bool {
```

Comment:
rename to `supports_initial_routing_sync` for consistency? & maybe not here, but maybe `InitialRoutingSync` about how it's a "known" feature to us but we won't send a complete routing dump to nodes that advertise this feature (iiuc)?

Reply:
Right, maybe leave this. Its a request. I think many other nodes don't bother to honor it anymore because they don't properly shape their outbound message queue, but we do.